<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Getting Started</title>

<script src="site_libs/header-attrs-2.8/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/yeti.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Home</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="Hi-C_workshop_intro.html">Getting Started</a>
</li>
<li>
  <a href="Hi-C_workshop_processing.html">Mapping and Filtering</a>
</li>
<li>
  <a href="Hi-C_workshop_formatting.html">Creating the Matrix</a>
</li>
<li>
  <a href="Hi-C_workshop_visualisation.html">Visualisation</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Getting Started</h1>

</div>


<head>
<script src="https://kit.fontawesome.com/ece750edd7.js" crossorigin="anonymous"></script>
</head>
<hr />
<p><img src="images/header.png" /></p>
<div id="overview" class="section level3">
<h3>Overview</h3>
<p>After a brief introduction, I will cover the steps involved in the processing of Hi-C data, from raw fastq reads into a contact matrix, then we will jump straight into visualizing data and exploring features of interest.</p>
<p><br></p>
</div>
<div id="getting-started-with-linux" class="section level3">
<h3>Getting Started with Linux</h3>
<p>We are using the Linux command line to run most of the tools we use today. If you are new to Linux please complete the Intro to Command Line <a href="http://bifx-core.bio.ed.ac.uk/hyweldd/training/Bioinformatics_on_the_Command_Line/">Workshop</a>.</p>
<p><br></p>
<div id="logging-in" class="section level4">
<h4>Logging in</h4>
<p>We have several servers that you can login to. For the purpose of this practical we will use bifx-core3. No matter where you login, you will have access to the same files and programs. If not on the University network, this requires you to use a VPN.</p>
<p>There are several options to login to our machines. You can use the <strong>Terminal</strong> app on a Mac or equivalant <strong>Command Prompt</strong> or MobaXTerm on Windows. Login via <strong>X2G0</strong> if you want a graphical interface.</p>
<p>To login via command line: ssh <a href="mailto:USER@bifx-core3.bio.ed.ac.uk" class="email">USER@bifx-core3.bio.ed.ac.uk</a></p>
<p>Login with your user credentials for our servers</p>
<p>If you are using MobaXTerm, an alternative way of logging in to the server is shown in the MobaXTerm demo.</p>
<p>Once you have typed in your password, you should see some welcome text and a prompt that looks something like this:</p>
<p>[USERNAME]<span class="citation">@bifx-core3</span>:~$</p>
<!--Alternatively you can use **RStudio** which is an environment built for R programming but also gives you access to the command line terminal, your file structure and several other graphical features. 

1. Open this link in a new window: [https://bifx-rstudio.bio.ed.ac.uk/](https://bifx-rstudio.bio.ed.ac.uk/)
2. Login with your user credentials for our servers
3. Familiarise yourself with the different panels and locate the Terminal. -->
<p><br></p>
</div>
</div>
<div id="creating-a-web-directory" class="section level3">
<h3>1. Creating A Web Directory</h3>
<p>In order to view files created on the server, we need to create a public_html directory.</p>
<p>After logging in you should be in your $HOME directory, check with;</p>
<pre class="bash"><code>pwd</code></pre>
<p>This should show the PATH of your present working directory, which should now be your home directory as you have just logged in. You can return to this place at any time using the change directory command.</p>
<pre class="bash"><code>cd</code></pre>
<p>You have permissions to create files and directories under your home folder. Lets create some now which we will use later on.</p>
<pre class="bash"><code>mkdir ~/public_html
mkdir ~/public_html/TMP</code></pre>
<p>Here we have used the absolute path name for each directory using ~/ as a shortcut for your $HOME directory. Nested directories are separated by the forward slash ‘/’ sign.</p>
<p>As you have created ~/public_html, contents of this directory are available online with any web browser</p>
<p>To see it enter the following URL, changing yourUserName to what ever your username is.</p>
<p><a href="http://bifx-core3.bio.ed.ac.uk/~yourUserName" class="uri">http://bifx-core3.bio.ed.ac.uk/~yourUserName</a></p>
<p>For some of you, this may be found here instead;</p>
<p><a href="https://bifx-core3.bio.ed.ac.uk/Public/yourUserName/" class="uri">https://bifx-core3.bio.ed.ac.uk/Public/yourUserName/</a></p>
<p><br></p>
</div>
<div id="data" class="section level3">
<h3>2. Data</h3>
<p>First, make a new directory for this tutorial and move into that directory. Then link the directory to your public html folder as we are going to make everything public in this tutorial.</p>
<pre class="bash"><code>cd 
mkdir Hi-C_workshop
cd Hi-C_workshop
ln -s $PWD ~/public_html/</code></pre>
<p>As Hi-C processing is slow, we are using some test data from the Dixon et al. 2012 <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3356448/">paper</a>.</p>
<p>These contain a small subsets of reads (~200,000).</p>
<p>Let’s copy the data into our project folder, we are copying multiple folders and files so need the recursive ‘-r’ argument.</p>
<pre class="bash"><code>cp -r /homes2/drober19/hic_workshop/test_data .</code></pre>
<p>The data is also available <a href="http://bifx-core3.bio.ed.ac.uk/Dan/Training/HiC_Workshop/test_data/">here</a></p>
<p>When you receive data from a sequencing centre the file should also be provided with an alphanumeric string known as an <strong>md5 checksum</strong>. We can think of this as a files passport or fingerprint and use it to verify our data and ensure it wasn’t corrupted or truncated during download. The md5 checksums for these files are below. Lets check that now using the <code>md5sum</code> command:</p>
<table>
<colgroup>
<col width="60%" />
<col width="40%" />
</colgroup>
<thead>
<tr class="header">
<th>md5 checksum</th>
<th>filename</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>94662045313249dec4d74f1eb337ddea</td>
<td>test_data/dixon_2M_2/SRR400264_01_R1.fastq</td>
</tr>
<tr class="even">
<td>43fcb38e4c64146a4ff7872176f7b698</td>
<td>test_data/dixon_2M_2/SRR400264_01_R2.fastq</td>
</tr>
<tr class="odd">
<td>41842bf0dcc77fb9cd4107b103d2588e</td>
<td>test_data/dixon_2M/SRR400264_00_R1.fastq</td>
</tr>
<tr class="even">
<td>09db06e20633213743a3081f0a3121a9</td>
<td>test_data/dixon_2M/SRR400264_00_R2.fastq</td>
</tr>
</tbody>
</table>
<pre class="bash"><code>md5sum test_data/*/* &gt; md5
cat md5 # prints out the contents of md5
#To check the files and md5 sums match at any time
md5sum -c md5 </code></pre>
<p><br></p>
</div>
<div id="higlass" class="section level3">
<h3>HiGlass</h3>
<p>We will be using HiGlass later on;</p>
<p>Again, you’ll need a VPN to access the local server if you’re using your own laptop;</p>
<p><a href="http://bifx-core3.bio.ed.ac.uk:32771/app" class="uri">http://bifx-core3.bio.ed.ac.uk:32771/app</a></p>
<p>Or use the <a href="https://higlass.io/app">official HiGlass site</a>.</p>
<p><br> <br></p>
<div class="key-points">
<h2>
<i class="fas fa-thumbtack"></i>Key Aims:
</h2>
<div id="section" class="section level2">
<h2></h2>
<ul>
<li>Be able to login to the bifx servers</li>
<li>Create a personal web directory</li>
<li>Create a project directory for fastq files</li>
</ul>
</div>
</div>
<p><br></p>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
