<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Creating the Matrix</title>

<script src="site_libs/header-attrs-2.21/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/yeti.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Home</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="Hi-C_workshop_intro.html">Getting Started</a>
</li>
<li>
  <a href="Hi-C_workshop_processing.html">Mapping and Filtering</a>
</li>
<li>
  <a href="Hi-C_workshop_formatting.html">Creating the Matrix</a>
</li>
<li>
  <a href="Hi-C_workshop_visualisation.html">Visualisation</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Creating the Matrix</h1>

</div>


<head>
<script src="https://kit.fontawesome.com/ece750edd7.js" crossorigin="anonymous"></script>
</head>
<hr />
<div id="the-matrix" class="section level3">
<h3>The Matrix</h3>
<p>Before visualization of our matrix lets go over the basics of a
contact map.</p>
<p>Areas of high intensity mean lots of interactions in that location,
the matrix allows us to see interactions across the whole genome, both
short and long range.</p>
<p><img src="images/cis_trans.png" /></p>
<p>Here we have a matrix showing chromosomes 1, 2 and 3, you can clearly
see chromosome territories, i.e. chromosomes are interacting with
themselves (cis) with fewer interchromosomal (trans) interactions.</p>
<p>Here we are zooming into chromosome 3, as expected there are lots of
local interactions close to the diagonal, these regions are close in the
linear sense so are more likely to interact which each other. As we move
further away from the diagonal there are fewer interactions as these
regions are further apart.</p>
<p><img src="images/simple.png" /></p>
<p>Note; the lower left half of the matrix is just a mirror image of the
the upper right.</p>
<p><br></p>
<div id="normalisation" class="section level4">
<h4>Normalisation</h4>
<p>Hi-C data contains systematic biases which need to be balanced. This
is down to both technical bias of sequencing/mapping and biological
artifacts such as GC content. Here the ICE method of normalisation is
applied to a raw hic matrix.</p>
<p><img src="images/norm.png" /></p>
<p><br></p>
</div>
</div>
<div id="file-formats" class="section level3">
<h3>File Formats</h3>
<p>After processing we should now have paired contacts that contain only
valid interactions. This allows us to plot a matrix showing points of
interaction. As mentioned previously, HiC-Pro has given us a matrix but
it is not in the format we need. There are currently two principle
formats used for Hi-C, these are each tied to their respective tools
(.hic for Juicebox &amp; .cool for HiGlass). Both formats/tools are
supported by 4D Nucleome program.</p>
<p><br></p>
<p><img src="images/4dn.png" /></p>
<p>‘A significant portion of bioinformatics is just converting data from
one format to another.’ Jason Stajich</p>
<!--Matrix (triplet sparse format)
Two interacting bins and an interaction frequency.

dixon_out/hic_results/matrix/dixon_2M/raw/20000/dixon_2M_20000.matrix
28 113593 1
41 41 1
41 62970 1

These can be plotted directly, here with HiCPlotter;

#python /usr/local/HiCPlotter-master/HiCPlotter.py -tri 1 -f dixon_out/hic_results/matrix/dixon_2M/iced/1000000/dixon_2M_1000000_iced.matrix -bed dixon_out/hic_results/matrix/dixon_2M/raw/1000000/dixon_2M_1000000_abs.bed -n dixon_2M_ICE_1mb_chr1 -chr chr1 -o hicplotter_out_r1mb -r 1000000
-->
<p><br></p>
<div id="valid-pairs" class="section level4">
<h4>Valid Pairs</h4>
<p>These are the pairs of coordinates for valid interactions only.</p>
<p>dixon_out/hic_results/data/dixon_2M/dixon_2M.allValidPairs</p>
<p>This is what we want to work with, to convert into out required
format.</p>
<pre class="bash"><code>head dixon_out/hic_results/data/dixon_2M/dixon_2M.allValidPairs</code></pre>
<p><br></p>
</div>
<div id="formatting-with-cooler" class="section level4">
<h4>Formatting with Cooler</h4>
<p>As we’re going to be using HiGlass, lets convert the valid pairs into
the cool format, the cooler tool can do this.</p>
<p>We can either do this by running multiple cooler commands directly or
making use of the HiC-Pro script hicpro2higlass.sh</p>
<p>Using HiC-Pro Utilities This can be achieved by running;</p>
<p>Get the chromsizes file</p>
<pre class="bash"><code>cp /homes2/drober19/hic_workshop/chrom_hg19.sizes .</code></pre>
<p>Create an output directory</p>
<pre class="bash"><code>mkdir cooler_out</code></pre>
<p>Note the resolution ‘-r’ is set to 10kb for the cool file, this will
also give us a .mcool file with multiple resolutions. -n matrix is
balanced (normalised)</p>
<pre class="bash"><code>/library/software/misc_containers/containers/hicpro/3.1.0/hicpro_3.1.0_ubuntu.img hicpro2higlass.sh -i dixon_out/hic_results/data/dixon_2M/dixon_2M.allValidPairs -r 10000 -c chrom_hg19.sizes -p 2 -n -o cooler_out &amp;

/library/software/misc_containers/containers/hicpro/3.1.0/hicpro_3.1.0_ubuntu.img hicpro2higlass.sh -i dixon_out/hic_results/data/dixon_2M_2/dixon_2M_2.allValidPairs -r 10000 -c chrom_hg19.sizes -p 2 -n -o cooler_out &amp;</code></pre>
<p><br></p>
</div>
<div id="using-cooler" class="section level4">
<h4>Using Cooler</h4>
<p>Or if you want to do this by step by step, see <a
href="https://github.com/mirnylab/cooler-binder/blob/master/cooler_cli.ipynb"
class="uri">https://github.com/mirnylab/cooler-binder/blob/master/cooler_cli.ipynb</a>.
Note this still requires some wrangling of the HiC-Pro output.</p>
<p>We should now all have a set of balanced matrices in the .cool/.mcool
formats.</p>
<p>This mcool file can be loaded into HiGlass, unfortunately this can
only been done by an admin but I’ve already added these files for
demonstration;</p>
<p><img src="images/dixon_sparse.png" /> Remember this was just a subset
of reads so we see a sparse matrix. We will play more with HiGlass later
on using some different data.</p>
<!--You can turn this off in HiGlass, by clicking options for a matrix then Configure Series/Transforms/None.

Alternative methods exist, but the differences are opaque; ICE/KR/VC Comparison -->
<p><br></p>
</div>
<div id="formatting-with-juicer-tools" class="section level4">
<h4>Formatting with Juicer Tools</h4>
<p>An alternative visualisation tool is Juicebox, this uses the .hic
format which is created using Juicer or Juicer Tools, thankfully HiC-Pro
also comes with a script that creates this Juicebox file from our
HiC-Pro output; hicpro2juicebox.sh</p>
<p>Create another output directory</p>
<pre class="bash"><code>mkdir juicer_tools_out</code></pre>
<p>We can create .hic files by running;</p>
<pre class="bash"><code>sh PATH_TO_SCRIPT/hicpro2juicebox.sh -i dixon_out/hic_results/data/dixon_2M/dixon_2M.allValidPairs
 -j /library/software/misc_containers/containers/hicpro/3.1.0/juicer_tools_2.13.06.jar -g hg19 -o juicer_tools_out &amp;

sh PATH_TO_SCRIPT/hicpro2juicebox.sh -i dixon_out/hic_results/data/dixon_2M/dixon_2M.allValidPairs
 -j /library/software/misc_containers/containers/hicpro/3.1.0/juicer_tools_2.13.06.jar -g hg19 -o juicer_tools_out &amp;</code></pre>
<p>Once we have the .hic files we can load these into directly into
Juicebox, we have a choice between the web version and a local GUI which
can be run on your own machine (much like IGV). For simplicity, today we
will just use the webtool (Note the GUI version does offer some
additional functionality).</p>
<p>This can be access here; <a href="https://aidenlab.org/juicebox/"
class="uri">https://aidenlab.org/juicebox/</a></p>
<p><img src="images/jb_dixon.png" /></p>
<p>To load your data select ‘Load Map’ from the top toolbar and then
‘Local File’ to navigate to your file. In this case you will need to
download the file from your public_html space, or copy this from the
server.</p>
<p>e.g. <a
href="https://bifx-core3.bio.ed.ac.uk/~drober19/Hi-C_workshop/juicer_tools_out/"
class="uri">https://bifx-core3.bio.ed.ac.uk/~drober19/Hi-C_workshop/juicer_tools_out/</a></p>
<p>Again, this is just a subset of the original experiment, see you if
you can find the full dataset within the Juicebox data archive (reminder
this is from Dixon et al. human IMR90 fibroblasts).</p>
<p>As we’ve seen, one of the main advantages of Juicebox is the ease of
loading our own data, there are advantages to both platforms, personally
I prefer HiGlass which we will delve further into shortly.</p>
<p><br></p>
</div>
</div>
<div id="hicexplorer" class="section level3">
<h3>HiCExplorer</h3>
<p>Once we have matrix files (cool / hic) we can use these as input for
other tools for analysis, <a
href="https://hicexplorer.readthedocs.io/en/latest/index.html">HiCExplorer</a>
is a tool with a lot of useful functions for further analysis. It
accepts multiple file formats :)</p>
<p>Lets compare our matrices using ‘hicPlotDistVsCounts’, this allows a
quick comparison between multiple matrices of the Hi-C counts
‘enrichment’ at different genomic distances. Biological replicates
should display the exact same distribution</p>
<p>This creates a distance vs. Hi-C counts plot;</p>
<pre class="bash"><code>hicPlotDistVsCounts -m cooler_out/dixon_2M.cool cooler_out/dixon_2M_2.cool -o counts_vs_dist.png </code></pre>
<p><img src="images/counts_vs_dist_10kb_dixon.png" /></p>
<pre><code>How do our samples look? Remember these are small subsets.</code></pre>
<p><img src="images/counts_vs_dist_example.png" /></p>
<p>Above is an example where replicates from condition one are not well
correlated, but are distinct from the second condition. The second are
well correlated and display enriched long-range contacts. (All are
smoother as these were binned at 50kb)</p>
<p><br></p>
<div class="key-points">
<h2>
<i class="fas fa-thumbtack"></i>Key Aims:
</h2>
<div id="section" class="section level2">
<h2></h2>
<ul>
<li>Interpreting a matrix</li>
<li>Converting file formats</li>
<li>Downstream Tools</li>
</ul>
</div>
</div>
<p><br></p>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
